### ISSUE TITLE
PWM?

### ISSUE THREAD

**Issue by Alfatross** [greatscottgadgets/greatfet-tutorials#10]
I'm not sure but is there anyway to set the greatfet to Pulse Width Modulation? if so could I get a code example?

**Comment by ktemkin** [comment]
This isn't in the current firmware release, but the next release will have a "pattern generator" peripheral that will provide support for PWM. That should be out really soon; once it is, we'll hopefully provide a code example. :)

One thing I think would be nice would be a simple "cookbook"-style wiki page -- one that collects neat snippets meant to do various tasks on Azalea / GF1 boards.

**Comment by Alfatross** [comment]
Yes I agree! There is a module that talks about downloading Bitstring but when I imported it, it only runs on python 2.7 not python3 which is listed in one of the tutorials however my computer class that I am now in says the only libraries that are downloaded with pip3 will only be able to use libraries with python 3 since the whole module of the GreatFet is installed with pip3 that only means libraries with the installation of pip3 can be used. Either way there should be a cookbook of code snippets like you said but not only that but as a newbie that I myself like to use python when writing code I believe you all should look at your competition with other Mirco-Controllers like Ardunio and Raspberry Pi and I do believe the Ardunio has the most use and capability. I would also like to see Analog/Digital GPIO pins or perhaps a way to use the existing capabilities to be able to act like one! I am just giving feedback because I do believe the Greatfet could grow into the market that could set itself apart from others which I am hoping to see!
